### PR TITLE
Docs: Fix docs for subnet allocation pool

### PR DESCRIPTION
**Pull request by jtopjian**
For #764

### CODE REVIEW

**Comment by jtopjian** [comment]
recheck

**Comment by kayrus** [comment]
Two flapping issues:

This is probably related to the race condition I noticed in the #590

```
--- FAIL: TestAccOpenStackImagesV2ImageDataSource_basic (36.79s) 
    testing.go:568: Step 0 error: errors during apply:

        Error: Error wrong checksum: got "2c40466809f9f984b13859abeb0fc210", expected "e6eec18d8db5c39829a35b9fea20d8d9"

          on /tmp/tf-test254942569/main.tf line 14:    
          (source code not available)
```

The one below is failing because of lower/upper case issues:

```
2019/06/06 05:43:54 [WARN] Provider "openstack" produced an invalid plan for openstack_networking_secgroup_rule_v2.secgroup_rule_1, but we are tolerating it because it is using the legacy plugin SDK.
    The following problems may be the cause of any confusing errors from downstream operations:
      - .remote_ip_prefix: planned value cty.StringVal("2001:558:fc00::/39") does not match config value cty.StringVal("2001:558:FC00::/39")

2019/06/06 05:43:55 [DEBUG] openstack_networking_secgroup_rule_v2 create options: rules.CreateOpts{Direction:"ingress", Description:"", EtherType:"IPv6", SecGroupID:"213a601e-d00c-498f-ab15-712c96580632", PortRangeMax:22, PortRangeMin:22, Protocol:"tcp", RemoteGroupID:"", RemoteIPPrefix:"2001:558:FC00::/39", ProjectID:""}
2019/06/06 05:43:55 [DEBUG] OpenStack Request URL: POST http://192.168.0.27:9696/v2.0/security-group-rules
2019/06/06 05:43:55 [DEBUG] OpenStack Request Headers:
Accept: application/json
Content-Type: application/json
User-Agent: Terraform/0.12.0 gophercloud/2.0.0
X-Auth-Token: ***
2019/06/06 05:43:55 [DEBUG] OpenStack Request Body: {
  "security_group_rule": {
    "direction": "ingress",
    "ethertype": "IPv6",
    "port_range_max": 22,
    "port_range_min": 22,
    "protocol": "tcp",
    "remote_ip_prefix": "2001:558:FC00::/39",    
    "security_group_id": "213a601e-d00c-498f-ab15-712c96580632"
  }
}
2019/06/06 05:43:55 [DEBUG] OpenStack Response Code: 201
2019/06/06 05:43:55 [DEBUG] OpenStack Response Headers:
Content-Length: 523
Content-Type: application/json
Date: Thu, 06 Jun 2019 05:43:55 GMT
X-Openstack-Request-Id: req-c952fd4a-04b7-47db-88de-a3fcf9eac323 
2019/06/06 05:43:55 [DEBUG] OpenStack Response Body: {
  "security_group_rule": {
    "created_at": "2019-06-06T05:43:55Z",
    "description": "",
    "direction": "ingress",
    "ethertype": "IPv6",
    "id": "1364db97-31b3-492d-b960-3ffcf5f67ffb",
    "port_range_max": 22,
    "port_range_min": 22,
    "project_id": "17778b8e6cb84005af48bace2f0e7682",
    "protocol": "tcp",
    "remote_group_id": null,
    "remote_ip_prefix": "2001:558:fc00::/39",    
    "revision_number": 0,
    "security_group_id": "213a601e-d00c-498f-ab15-712c96580632",
    "tenant_id": "17778b8e6cb84005af48bace2f0e7682",
    "updated_at": "2019-06-06T05:43:55Z"
  }
}
2019/06/06 05:43:55 [DEBUG] Created openstack_networking_secgroup_rule_v2 1364db97-31b3-492d-b960-3ffcf5f67ffb: &rules.SecGroupRule{ID:"1364db97-31b3-492d-b960-3ffcf5f67ffb", Direction:"ingress", Description:"", EtherType:"IPv6", SecGroupID:"213a601e-d00c-498f-ab15-712c96580632", PortRangeMin:22, PortRangeMax:22, Protocol:"tcp", RemoteGroupID:"", RemoteIPPrefix:"2001:558:fc00::/39", TenantID:"17778b8e6cb84005af48bace2f0e7682", ProjectID:"17778b8e6cb84005af48bace2f0e7682"}

--- FAIL: TestAccComputeV2SecGroup_lowerCaseCIDR (2.22s)
    testing.go:568: Step 0 error: Check failed: Check 2/2 error: openstack_compute_secgroup_v2.sg_1: Attribute 'rule.3862435458.cidr' not found
```

File: website/docs/r/networking_subnet_v2.html.markdown [hunk 1]
```diff
@@ -64,12 +64,16 @@ The following arguments are supported:
     create a subnet for another tenant. Changing this creates a new subnet.
 
 * `allocation_pools` - (**Deprecated** - use `allocation_pool` instead)
-    An array of sub-ranges of CIDR available for dynamic allocation to ports.
-    The allocation_pools object structure is documented below.
-
-* `allocation_pool` - (Optional) An array of sub-ranges of CIDR available for
-    dynamic allocation to ports. The allocation_pool object structure is
-    documented below.
+    A block declaring the start and end range of the IP addresses available for
```
Comment (keiranmraine) [inline website/docs/r/networking_subnet_v2.html.markdown]: Not convinced the description of the deprecated item should be modified

File: website/docs/r/networking_subnet_v2.html.markdown [hunk 2]
```diff
@@ -64,12 +64,16 @@ The following arguments are supported:
...
+    use with DHCP in this subnet.
+    The `allocation_pools` block is documented below.
+
+* `allocation_pool` - (Optional) A block declaring the start and end range of
```
Comment (keiranmraine) [inline website/docs/r/networking_subnet_v2.html.markdown]: This is much clearer now, thanks.

File: website/docs/r/networking_subnet_v2.html.markdown (hunk 1 above)
Comment (jtopjian) [reply website/docs/r/networking_subnet_v2.html.markdown]: I can see your point. It's deprecated, sure, but it's just as incorrect as the non-deprecated item.

**Comment by jtopjian** [comment]
@kayrus regarding the test failures, I think another possibility is the underlying host that the test is run on OpenLab utilizes several different organizations who contribute space on their OpenStack clouds. I have an open issue about this somewhere at OpenLab.

**Comment by jtopjian** [comment]
@keiranmraine Thank you for the review!

**Pull request merged**